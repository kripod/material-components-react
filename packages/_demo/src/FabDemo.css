.fab-absolute {
  position: absolute;
  bottom: 1rem;
  right: 1rem;
}
