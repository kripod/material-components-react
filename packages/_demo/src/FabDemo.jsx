import Fab from '@react-material/fab';
import React from 'react';
import './FabDemo.css';

const FabDemo = () =>
  <div>
    <h1>FAB (Floating Action Button)</h1>

    <Fab className="fab-absolute">
      <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24">
        <path d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z" />
      </svg>
    </Fab>

    <section>
      {[true, false].map(isEnabled =>
        <fieldset key={isEnabled} disabled={!isEnabled}>
          <legend>{isEnabled ? 'Enabled' : 'Disabled'}</legend>

          <Fab aria-label="Favorite">favorite_border</Fab>
          <Fab mini aria-label="Favorite">favorite_border</Fab>
          <Fab plain aria-label="Favorite">favorite_border</Fab>
          <Fab mini plain aria-label="Favorite">favorite_border</Fab>
        </fieldset>,
      )}
    </section>
  </div>
;

export default FabDemo;
