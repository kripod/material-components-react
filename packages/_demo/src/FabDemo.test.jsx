import React from 'react';
import renderer from 'react-test-renderer';
import FabDemo from './FabDemo';

it('renders correctly', () => {
  const tree = renderer.create(<FabDemo />).toJSON();
  expect(tree).toMatchSnapshot();
});
