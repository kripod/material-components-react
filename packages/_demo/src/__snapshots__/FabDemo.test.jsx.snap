exports[`test renders correctly 1`] = `
<div>
  <h1>
    FAB (Floating Action Button)
  </h1>
  <button
    className="mdc-fab "
    style={
      Object {
        "bottom": "1rem",
        "position": "absolute",
        "right": "1rem",
      }
    }>
    <span
      className="mdc-fab__icon">
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg">
        <path
          d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.39-.39-1.02-.39-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z" />
      </svg>
    </span>
  </button>
  <section>
    <fieldset
      disabled={false}>
      <legend>
        Enabled
      </legend>
      <button
        aria-label="favorite_border"
        className="mdc-fab material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--mini material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--plain material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--mini mdc-fab--plain material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
    </fieldset>
    <fieldset
      disabled={true}>
      <legend>
        Disabled
      </legend>
      <button
        aria-label="favorite_border"
        className="mdc-fab material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--mini material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--plain material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
      <button
        aria-label="favorite_border"
        className="mdc-fab mdc-fab--mini mdc-fab--plain material-icons ">
        <span
          className="mdc-fab__icon">
          favorite_border
        </span>
      </button>
    </fieldset>
  </section>
</div>
`;
